bulk/management/commands/data_quality.py

```diff
@@ -0,0 +1,343 @@
+import os
+import csv
+import json
+import datetime
+import tempfile
+import zipfile
+import uuid
+import boto3
+import base62
+from django.core.management.base import BaseCommand
+from django.db.models import F, Count, Avg
+from openstates_metadata import STATES_BY_NAME
+from openstates.data.models import (
+    LegislativeSession,
+    Bill,
+    BillAbstract,
+    BillAction,
+    BillTitle,
+    BillIdentifier,
+    RelatedBill,
+    BillSponsorship,
+    BillDocument,
+    BillVersion,
+    BillDocumentLink,
+    BillVersionLink,
+    BillSource,
+    VoteEvent,
+    PersonVote,
+    VoteCount,
+    VoteSource,
+)
+from utils.common import abbr_to_jid
+from utils.orgs import get_chambers_from_abbr
+from collections import defaultdict
+from statistics import mean
+
+# Loads the global bill array with all bills from given state and session to use
+#   when creating the json
+def load_bills(state, session):
+    sobj = LegislativeSession.objects.get(
+        jurisdiction_id=abbr_to_jid(state), identifier=session
+    )
+    bills = Bill.objects.filter(legislative_session=sobj).prefetch_related("actions", "sponsorships", "votes", "votes__counts", "sources", "documents", "versions", "votes__votes")
```

Comment:
can do this in one query via joins

Bill.objects.filter(legislative_session__jurisdiction_id=abbr_to_jid(state), identifier=...)

Reply:
Now uses only one query.